### PR TITLE
gh-80448: argparse: Fix IndexError on store_true action

### PR DESCRIPTION
**Pull request by shihai1991**
https://bugs.python.org/issue36267

* Issue: gh-80448

### CODE REVIEW

**Comment by shihai1991** [comment]
cc @rhettinger

File: Misc/NEWS.d/next/Library/2019-09-03-15-45-19.bpo-36267.z42Ex7.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Fix IndexError in argparse when instance of  :class:`ArgumentParser` parse args like '-a=' what have a 'store_true' action.
```
Comment (remilapeyre) [inline Misc/NEWS.d/next/Library/2019-09-03-15-45-19.bpo-36267.z42Ex7.rst]:
```suggestion
Fix IndexError in :class:`argparse.ArgumentParser` when a `store_true` action is given an explicit argument.
```

**Review by remilapeyre — APPROVED** [review]

File: Misc/NEWS.d/next/Library/2019-09-03-15-45-19.bpo-36267.z42Ex7.rst (hunk 1 above)
Comment (shihai1991) [reply Misc/NEWS.d/next/Library/2019-09-03-15-45-19.bpo-36267.z42Ex7.rst]: Thanks for your comment. your comment looks more simple and exact.

PS: I have forget this PR's detail in fact~.

File: Lib/argparse.py [hunk 2]
```diff
@@ -1897,7 +1897,9 @@ def consume_optional(start_index):
                     # arguments, try to parse more single-dash options out
                     # of the tail of the option string
                     chars = self.prefix_chars
-                    if arg_count == 0 and option_string[1] not in chars:
+                    if (arg_count == 0 and
+                        option_string[1] not in chars and
+                        explicit_arg != ''):
```
Comment (hauntsaninja) [inline Lib/argparse.py]: Optional style nit (increases visual separation between if condition and body)
```suggestion
                    if (
                        arg_count == 0
                        and option_string[1] not in chars
                        and explicit_arg != ''
                    ):
```

File: Misc/NEWS.d/next/Library/2019-09-03-15-45-19.bpo-36267.z42Ex7.rst [hunk 3]
```diff
@@ -0,0 +1 @@
+Fix IndexError in :class:`argparse.ArgumentParser` when a `store_true` action is given an explicit argument.
```
Comment (hauntsaninja) [inline Misc/NEWS.d/next/Library/2019-09-03-15-45-19.bpo-36267.z42Ex7.rst]: Fix RST markup
```suggestion
Fix IndexError in :class:`argparse.ArgumentParser` when a ``store_true`` action is given an explicit argument.
```

**Review by hauntsaninja — APPROVED** [review]
Looks good! Had two nits, although the style one can be ignored cc @JelleZijlstra

**Review by JelleZijlstra — APPROVED** [review]

**Comment by JelleZijlstra** [comment]
Planning to merge this after CI passes, and backport to the bugfix branches.

**Pull request merged**